gnupg/_parsers.py

```diff
@@ -1027,6 +1033,9 @@ def sub(self, args):
         subkey = [args[4], args[11]]
         self.curkey['subkeys'].append(subkey)
 
+    def rvk(self, args):
+        self.curuid == None
```

Comment:
I don't see this revocation key packet in your error example at #158.

Why are you reseting the curuid when you find a revocation key?